app/mailers/application_mailer.rb

```diff
@@ -36,4 +37,27 @@ def generate_unsubscribe_token(id, email_type)
   def use_custom_host
     ActionMailer::Base.default_url_options[:host] = Settings::General.app_domain
   end
+
+  def perform_deliveries?
```

Comment:
I'm not sure why but something is behaving strangely or I'm just seeing things.

On `main`:

```ruby
[1] pry(main)> ApplicationMailer.perform_deliveries?
NoMethodError: undefined method `perform_deliveries?' for ApplicationMailer:Class
[2] pry(main)> ApplicationMailer.new.perform_deliveries?
NoMethodError: undefined method `perform_deliveries?' for #<ApplicationMailer:0x000000000273f8>
```

as it should, since there's only the accessor [`perform_deliveries`](https://api.rubyonrails.org/classes/ActionMailer/Base.html):

```ruby
> ApplicationMailer.perform_deliveries
=> true
> ApplicationMailer.new.perform_deliveries
=> true
```

as you can see, this is the same value for both class and instance, because it's defined as a [mattr_accessor](https://api.rubyonrails.org/classes/Module.html#method-i-mattr_accessor)

but, if I switch to this branch, things start happening:

```ruby
[1] pry(main)> ApplicationMailer.perform_deliveries?
  User Exists? (132.5ms)  SELECT 1 AS one FROM "users" LIMIT $1  [["LIMIT", 1]]
Unset ENV variable: SMTP_ADDRESS.
Unset ENV variable: SMTP_PORT.
Unset ENV variable: SMTP_AUTHENTICATION.
Unset ENV variable: SMTP_USER_NAME.
Unset ENV variable: SMTP_PASSWORD.
Unset ENV variable: SMTP_DOMAIN.
ApplicationMailer#perform_deliveries?: processed outbound mail in 252.4ms
=> #<ActionMailer::Base::NullMail:0x00007fe390cbd5b0>
```

and everytime I call it it tells me it's processing something:

```ruby
[3] pry(main)> ApplicationMailer.perform_deliveries?
ApplicationMailer#perform_deliveries?: processed outbound mail in 0.2ms
=> #<ActionMailer::Base::NullMail:0x00007fe390c5ead8>
```

but if I call it as an instance method:

```ruby
> ApplicationMailer.new.perform_deliveries?
=> nil
```

then it works as expected.

For some reason the mailer is defining this as a class method as well. This also happens to other methods as well:

```ruby
> ApplicationMailer.use_custom_host
ApplicationMailer#use_custom_host: processed outbound mail in 0.3ms
=> #<ActionMailer::Base::NullMail:0x00007fc864dfc8c8>
```

🤷

Reply:
hmm interesting I also replicated this weird `processed outbound mail in 0.3ms` process.

Reply:
@maestromac If an instance method is defined on a mailer, calling that method on the mailer _class_ treats it as a mailer action. It's my [least favorite thing about ActionMailer](https://twitter.com/jamie_gaskins/status/396330864369336320).

Reply:
hahaha wow a tweet 2013. TIL!

Reply:
Yeah, I've been carrying that beef for a while :joy: